### ISSUE TITLE
Let's improve invalid ETA (Estimated Time to Arrival)

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#1845]
Please collect here all invalid ETAs with start and endpoints, and with an actual (real) time for the route. Then we can easily fix and improve it.

There are some assumptions in the code, ETA now ignores turns and traffic lights for example.

**Comment by sn0ot** [comment]
Realistically, how many independent routes do you need?

I can set up a survey link so users can anonymously enter data and I can periodically dump JSONs of aggregate data on this issue. 

Are you worried that traffic can skew the results? Or you would rather over estimate than the current under estimate?

Now's your chance to standardize the format of data, does this suffice?

```json
{
    "Route": {
        "start lat, long": "41.515449450036535, -74.39019167206676",
        "end lat, long": "41.44723791528648, -74.41565461594804",
        "invalid ETA": "13:09",
        "Actual (real) ETA": "13:12",
        "unusually heavy traffic present": "False"
    }
}
```

**Comment by biodranik** [comment]
That format can be improved (e.g. lat, lon can be stored separately). But that's not too important. The idea is to feed our C++ unit tests with real examples.

And please ignore traffic (e.g. assume that there is no traffic on the route). Until we find some source of traffic data, let's make our algorithms work better in the "normal" conditions.

**Comment by muralito** [comment]
What about collecting/calculating real driving/cycling times for each edge of the road network? OM already have every edge identified in map data with their speed/time/cost. When a user drives on each edge, OM can calculate and store the time spent, maybe including more data like day/hour. And then also an option to send the differences between your network data, and the data the user collected. Obviously all these options must be activated only if the user which wants to collaborate with OM enables them, and respecting the privacy.

**Comment by matheusgomesms** [comment]
On short routes in urban environment, ETA is really off by a large margin, from my experience (comparing to midnight/Sunday trips, without traffic). I think these penalties would adjust that a bit.

**Comment by patepelo** [comment]
> What about collecting/calculating real driving/cycling times for each edge of the road network? OM already have every edge identified in map data with their speed/time/cost. When a user drives on each edge, OM can calculate and store the time spent, maybe including more data like day/hour. And then also an option to send the differences between your network data, and the data the user collected. Obviously all these options must be activated only if the user which wants to collaborate with OM enables them, and respecting the privacy.

That could be an option, kind of offering to donate your traffic data anonymously (set to off by default). Could also have potential for KeepRight QA or similars.

**Comment by muralito** [comment]
Yes. Somehow you have to take care of several possibilities, e.g, someone driving above the OSM tagged speed limit, could be that in fact is speeding (the estimated time shouldn't be less than the minimum legal time), but it also could be that the limit has changed.

At least i saw two kind of uses, not mutually exclusive, for this data. 
1) Personal use:  data don't need to leave the device (but it could be shared between devices, and somehow persist between map updates). OM could adapt to every user personal history of previous driving times.
2) Collaborative use: data uploaded to OM, then processed and delivered to users in future map updates.
Both features shares a common implementation which collects data.

Maybe users could opt to collect data only when they are not speeding, or to collect always but only collaborate when they are not speeding...

Garmin nuvi units had a feature called myTrends, that Garmin says it learns from user habits and history, and improve suggestions and ETA as it is used.

**Comment by pastk** [comment]
Road crossings should be considered also for pedestrian and cycling routing.
